src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -2227,10 +2226,13 @@ ErrorIfUnsupportedShardDistribution(Query *query)
 
 
 /*
- * CoPartitionedTables checks if given two distributed tables have 1-to-1 shard
- * partitioning.
+ * CoPartitionedTables first checks for the shard count, if tables don't have
```

Comment:
Can't we start with something more generic sentence before givin into the details:
```CoPartitionedTables checks if given two distributed tables have 1-to-1 shard placement matching.```